### PR TITLE
BUG: Geocode Fails When results.value is None

### PR DESCRIPTION
**Pull request by rdmolony**
# Old Behaviour

I ran ` gpd.tools.geocode(addresses, provider="here", apikey="<KEY>")` on a `GeoSeries` called `addresses` and `geopandas` failed at:

```python-traceback
~/.cache/pypoetry/virtualenvs/drem-TFRFQYJy-py3.8/lib/python3.8/site-packages/geopandas/tools/geocoding.py in _prepare_geocode_result(results)
    170 
    171     for i, s in results.items():
--> 172         address, loc = s
    173 
```

With...

```python-traceback
TypeError: cannot unpack non-iterable NoneType object
```

So `geopandas.tools.geocode._prepare_geocode_result` failed when `s` in above example is `None` as it expects a Tuple such as `(`None, None)` that it can unpack.

# New Behaviour 
It now succeeds when `s` is `None` or `(None, None)` and two tests have been implemented to ensure that `geocode` passes in both of the above cases.

### CODE REVIEW

**Comment by rdmolony** [comment]
`test_append_file[GeoJSON-geojson]` seems to be causing AppVeyor to fail in 3.7 and 3.8

File: geopandas/tools/tests/test_geocode.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import numpy as np
```
Comment (martinfleis) [inline geopandas/tools/tests/test_geocode.py:1]: Can you move these to existing `geopandas/tests/test_geocode.py`?

**Review by martinfleis — COMMENTED** [review]
Thanks! Can you just move tests to the existing file?

File: geopandas/tools/tests/test_geocode.py (hunk 1 above)
Comment (rdmolony) [reply geopandas/tools/tests/test_geocode.py]: Sure thing :)

**Comment by martinfleis** [comment]
@rdmolony CI is failing due to style issues. Can you fix both flake8 and black issues? See https://geopandas.readthedocs.io/en/latest/community/contributing.html#style-guide-linting for more.

File: geopandas/tools/geocoding.py [hunk 2]
```diff
@@ -169,16 +169,22 @@ def _prepare_geocode_result(results):
     index = []
 
     for i, s in results.items():
-        address, loc = s
 
-        # loc is lat, lon and we want lon, lat
-        if loc is None:
+        if s is None:
             p = Point()
+            address = np.nan
```
Comment (jorisvandenbossche) [inline geopandas/tools/geocoding.py:175]:
```suggestion
            address = None
```

File: geopandas/tools/geocoding.py [hunk 3]
```diff
@@ -169,16 +169,22 @@ def _prepare_geocode_result(results):
...
-        if address is None:
-            address = np.nan
+            # loc is lat, lon and we want lon, lat
+            if loc is None:
+                p = Point()
+            else:
+                p = Point(loc[1], loc[0])
+
+            if address is None:
+                address = np.nan
```
Comment (jorisvandenbossche) [inline geopandas/tools/geocoding.py:187]:
```suggestion
```

I would leave it at `None` -> pandas also sees that as missing value (just as np.nan), and that's what we did in the past

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the fix! A small comment on None vs np.nan

**Comment by rdmolony** [comment]
Thanks for the help @martinfleis and @jorisvandenbossche!  I forgot to run`pre-commit install` after installing dev dependencies and so kept forgetting to blacken (& flake8) my code prior to pushing (and not for the first time)!  

Will submit a mini PR now to change contributing guidelines to prompt contributors to do this during creation of a conda dev env (so I for one don't do this again!)

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks a lot for the nice PR!

**Pull request merged**